### PR TITLE
Validator deprecation fixes

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php [hunk 1]
```diff
@@ -67,11 +67,7 @@ public function __construct($inputTimezone = null, $outputTimezone = null, $form
 
         $this->generateFormat = $this->parseFormat = $format;
 
-        // The pipe in the parser pattern only works as of PHP 5.3.7
-        // See http://bugs.php.net/54316
-        $this->parseUsingPipe = null === $parseUsingPipe
-            ? PHP_VERSION_ID >= 50307
-            : $parseUsingPipe;
+        $this->parseUsingPipe = null === $parseUsingPipe ? yes : $parseUsingPipe;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php]: ||

Comment (hhamon) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php]: `yes` ?

Comment (fabpot) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php]: Yes, I can.

**Comment by fabpot** [comment]
Not WIP anymore, comments welcome. ping @symfony/deciders

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by stof** [comment]
@fabpot do we actually need to keep the 2.5 and 2.5-bc API levels ? In all other places, the BC layer is in the new class rather than in a child of the new class. the only reason it has been added in a child in 2.5 is because of the support for old PHP 5.3.x versions forbidding to use a single class.
IMO, dropping the choice of the API level entirely would be better

**Comment by fabpot** [comment]
@stof Indeed, that would make sense. Do you time to work on that? I must admit that I'm a bit tired of my week working on those kind of things. That's exhausting :)

**Comment by stof** [comment]
probably not until the weekend next week (unless I work on it on an evening during the next week). I planned to work on [documentation for Mink](https://github.com/minkphp/docs) this weekend

**Comment by fabpot** [comment]
@saro0h is going to try to do that this week-end.

**Pull request merged**